include/adobe/contract_checks.hpp

```diff
@@ -1,169 +1,117 @@
 #ifndef ADOBE_CONTRACT_CHECKS_HPP
 #define ADOBE_CONTRACT_CHECKS_HPP
-#include <cstdint>
-#include <cstdio>
-#include <exception>
-#include <stdexcept>
-
-// ADOBE_MINIMAL_TRAP(): followed with a semicolon, expands to a
-// minimal statement that stops the program.
-#if !defined(__EMSCRIPTEN__) && (defined(__clang__) || defined(__GNUC__))
-#define ADOBE_MINIMAL_TRAP() __builtin_trap()
-#elif defined(_MSC_VER)
-#define ADOBE_MINIMAL_TRAP() __debugbreak()
-#else
-#define ADOBE_MINIMAL_TRAP() std::abort()
-#endif
 
-// Injects an inline definition of ::adobe::contract_violated that
-// stops the program in the most efficient known way, without any
-// diagnostic output.  This macro is only useful in your
-// ADOBE_CONTRACT_CHECKS_CONFIGURATION file.
-#define ADOBE_MINIMAL_INLINE_CONTRACT_VIOLATION_HANDLER()       \
-  namespace adobe {                                             \
-  [[noreturn]] inline void contract_violated(const char *const, \
-    int,                                                        \
-    const char *const,                                          \
-    std::uint32_t const,                                        \
-    const char *const) noexcept                                 \
-  {                                                             \
-    ADOBE_MINIMAL_TRAP();                                       \
-  }                                                             \
-  }
+#define INTERNAL_ADOBE_CONTRACT_VIOLATION_verbose 1
+#define INTERNAL_ADOBE_CONTRACT_VIOLATION_minimal 2
+#define INTERNAL_ADOBE_CONTRACT_VIOLATION_unsafe 3
 
-// Do not move this #include up; the included file needs to be able to use the above macros.
-#ifdef ADOBE_CONTRACT_CHECKS_CONFIGURATION
-#include ADOBE_CONTRACT_CHECKS_CONFIGURATION
-#endif
-
-#ifdef ADOBE_CONTRACT_VIOLATED_THROWS
-#define INTERNAL_ADOBE_CONTRACT_VIOLATED_NOEXCEPT
+// INTERNAL_ADOBE_CONTRACT_VIOLATION_BEHAVIOR is not a function-style
+// macro because it lets us give a better diagnostic on
+// misconfiguration.
+#ifndef ADOBE_CONTRACT_VIOLATION// Default is verbose
```

Comment:
```suggestion
#ifndef ADOBE_CONTRACT_VIOLATION // Default is verbose
```

Reply:
You can do that, but clang-format will put it back to the way it is.  If you want to change that you need to update the formatting file.

Reply:
Okay - I'll look at updating the format file (the current one is horrible).